### ISSUE TITLE
CSV headers with spaces are not handled correctly by gatsby-transformer-csv

### ISSUE THREAD

**Issue by tjscollins** [gatsbyjs/gatsby#5124]
## Description

If you try to parse CSV files that have headers containing spaces the data will be all be returned as null by graphql queries.  The problem seems to be that the GraphQL query substitutes underscores for spaces, but the same transformation is not being applied to the actual data.  The result is that you Query for `Header_One` but the data is stored in `node["Header One"]` and `node.Header_One` is null.

### Steps to reproduce

Parse any CSV file where the headers have spaces.  Try to query data from those fields that have spaces in the names, converted to underscores in the GraphQL query.  For Example:

```
{
    allTestCsv {
        edges {
            node {
                Header_One
            }
        }
    }
}
```

### Expected result

Query should return the data from the CSV file.

### Actual result

Any fields with spaces in the header return null values.

### Environment

* Gatsby version (`npm list gatsby`): gatsby@1.9.253
* gatsby-cli version (`gatsby --version`): 1.1.50
* Node.js version: v8.4.0
* Operating System: Ubuntu 17.10

**Comment by KyleAMathews** [comment]
Hmm sounds like a bug indeed! Would you be interested in putting together a PR to fix the problem? It sounds like applying the same change to the node key should solve the problem. Though actually a more general solution in core is worth thinking about as well as this would happen to any plugin.

**Comment by tjscollins** [comment]
The problem seems to stem from the fact that JS allows spaces in field names and GraphQL doesn't.  I see two possible solutions:

1. gatsby-transformer-csv should covert all spaces (and other GraphQL illegal characters) in field names to underscores, matching the GraphQl queries
2. Whatever code handles the GraphQL queries should be modified to convert between underscores in query params and spaces (or whatever other graphql illegal characters were used) in the field names as necessary.

Option 1 is easy enough to implement as a regex replace, though it would need some thorough testing.  Option 2 is probably better, but much more complicated.  There would have to be a map converting GraphQL query params to JS field names.  It's doable, but not simple, and may not be a good design choice.  I don't think I understand GraphQL or gatsby's architecture well enough to do Option 2 myself.  I could submit a PR for Option 1 though.

**Comment by tjscollins** [comment]
On reflection about my own use case I realized that there is a simpler solution than Option 2.  Rather than bothering with the complicated idea of mapping between GraphQL syntax and JS syntax, I thought it would better to implement Option 1, and document the issue, with the workaround for users who want to preserve the illegal characters (as I need to) of using the `{ noheaders: true }` option, which makes the headers into string data, sidestepping the issue.  

The PR I submitted implements this with a simple regex replace and an update to the README.
